Makefile

```diff
@@ -169,9 +169,33 @@ ifneq ($(CPU_ONLY), 1)
 	LIBRARY_DIRS += $(CUDA_LIB_DIR)
 	LIBRARIES := cudart cublas curand
 endif
-LIBRARIES += glog gflags protobuf leveldb snappy \
-	lmdb boost_system hdf5_hl hdf5 m \
-	opencv_core opencv_highgui opencv_imgproc
+
+LIBRARIES += protobuf m
+
+ifeq ($(USE_BOOST), 1)
+  LIBRARIES += boost_system
+endif
+ifeq ($(USE_GLOG), 1)
+  LIBRARIES += glog
+endif
+ifeq ($(USE_GFLAGS), 1)
+  LIBRARIES += gflags
+endif
+ifeq ($(USE_LEVELDB), 1)
+  LIBRARIES += leveldb
+endif
+ifeq ($(USE_SNAPPY), 1)
+    LIBRARIES += snappy
+endif
+ifeq ($(USE_LMDB), 1)
+   LIBRARIES += lmdb
+endif
+ifeq ($(USE_HDF5), 1)
+    LIBRARIES += hdf5_hl hdf5
+endif
+ifeq ($(USE_OPENCV), 1)
+	LIBRARIES += opencv_core opencv_highgui opencv_imgproc
+endif
 PYTHON_LIBRARIES := boost_python python2.7
```

Comment:
These changes require the users to update their Makefile.config. If they don't do that, the libraries are switched off by default which is inconsistent with CMakeLists.txt.